server/mux.go

```diff
@@ -67,127 +67,137 @@ func NewMux(opts MuxOpts, service Service) http.Handler {
+	router.GET("/chronograf/v1/sources/:id/kapacitors/:kid/rules", EnsureViewer(service.KapacitorRulesGet))
+	router.POST("/chronograf/v1/sources/:id/kapacitors/:kid/rules", EnsureEditor(service.KapacitorRulesPost))
 
-	router.GET("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", service.KapacitorRulesID)
-	router.PUT("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", service.KapacitorRulesPut)
-	router.PATCH("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", service.KapacitorRulesStatus)
-	router.DELETE("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", service.KapacitorRulesDelete)
+	router.GET("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", EnsureViewer(service.KapacitorRulesID))
+	router.PUT("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", EnsureEditor(service.KapacitorRulesPut))
+	router.PATCH("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", EnsureEditor(service.KapacitorRulesStatus))
+	router.DELETE("/chronograf/v1/sources/:id/kapacitors/:kid/rules/:tid", EnsureEditor(service.KapacitorRulesDelete))
 
 	// Kapacitor Proxy
-	router.GET("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", service.KapacitorProxyGet)
-	router.POST("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", service.KapacitorProxyPost)
-	router.PATCH("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", service.KapacitorProxyPatch)
-	router.DELETE("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", service.KapacitorProxyDelete)
+	router.GET("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", EnsureViewer(service.KapacitorProxyGet))
+	router.POST("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", EnsureEditor(service.KapacitorProxyPost))
+	router.PATCH("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", EnsureEditor(service.KapacitorProxyPatch))
+	router.DELETE("/chronograf/v1/sources/:id/kapacitors/:kid/proxy", EnsureEditor(service.KapacitorProxyDelete))
 
 	// Mappings
-	router.GET("/chronograf/v1/mappings", service.GetMappings)
+	router.GET("/chronograf/v1/mappings", EnsureViewer(service.GetMappings))
 
 	// Layouts
-	router.GET("/chronograf/v1/layouts", service.Layouts)
-	router.POST("/chronograf/v1/layouts", service.NewLayout)
+	router.GET("/chronograf/v1/layouts", EnsureViewer(service.Layouts))
+	router.POST("/chronograf/v1/layouts", EnsureEditor(service.NewLayout))
 
-	router.GET("/chronograf/v1/layouts/:id", service.LayoutsID)
-	router.PUT("/chronograf/v1/layouts/:id", service.UpdateLayout)
-	router.DELETE("/chronograf/v1/layouts/:id", service.RemoveLayout)
+	router.GET("/chronograf/v1/layouts/:id", EnsureViewer(service.LayoutsID))
+	router.PUT("/chronograf/v1/layouts/:id", EnsureEditor(service.UpdateLayout))
+	router.DELETE("/chronograf/v1/layouts/:id", EnsureEditor(service.RemoveLayout))
 
 	// Users associated with Chronograf
-	router.GET("/chronograf/v1/me", service.Me)
+	router.GET("/chronograf/v1/me", EnsureViewer(service.Me))
```

Comment:
this might have to be accessible to NoRole folk, and we might have to create a NoRole role. if a user is not Viewer by default, and i don't think they should be, i believe they'll still need to be able to receive something via `/me`

Reply:
good catch. I think we can just leave this route unauthorized.